### ISSUE TITLE
Stop packager when building/publishing in non-interactive mode

### ISSUE THREAD

**Issue by MaaKut** [expo/expo-cli#640]
## Description

I'm trying to build or publish the Expo-based application in non-interactive mode (required for CI/CD) like this:

```npx expo build:android --non-interactive```
or
```npx expo publish --non-interactive```

The problem: packager is started automatically and does not terminate on build completion, keeping the process running.

## Expected Behavior

It is expected, that once build/publish is complete, the packager also stops and the whole process terminates.

## Observed Behavior

The packager keeps running. This makes CI/CD think that the respective step has stalled, so it terminates the build process on timeout.

## Environment

```  Expo CLI 2.17.1 environment info:
    System:
      OS: Windows 7
    Binaries:
      Yarn: 1.9.4 - C:\Users\akiselev\AppData\Roaming\npm\yarn.CMD
      npm: 6.8.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.3.0.0 AI-182.5107.16.33.5199772```

**Comment by cruzach** [comment]
Hey @MaaKut, could you try passing `--max-workers 1` to `expo publish`? That's the suggested use in our [Setting Up CI guide](https://docs.expo.io/versions/latest/guides/setting-up-continuous-integration/#publish-new-builds)

Let me know if that works!
Charlie

**Comment by MaaKut** [comment]
Hello, @cruzach , thank you for looking into this.

> could you try passing --max-workers 1 to expo publish?
I've tried `npx expo publish --max-workers 1 --non-interactive` - same result, the process keeps on running.

**Comment by Exoshale** [comment]
Hey @MaaKut,

Does the packager complete successfully and not report that it completed or is it hanging during the publish process?

I'm trying to determine if I have a similar issue with expo publish and CI/CD or a new issue where the packager hangs (and never completes at all).

**Comment by MaaKut** [comment]
Hi, @Exoshale ,

It does look like the packager publishes my project successfully (at least, I can seen the changes on the devices), but after that it just keeps waiting for something until I `Ctrl+C` it.

**Comment by nemenos** [comment]
Same happens here, it's really hard to make a CI script working properly...
Thanks

**Comment by brentvatne** [comment]
hello! i'm taking over on expo-cli while @fson is away on pat leave. i believe this issue is resolved, i just ran `expo publish --non-interactive` and it worked as expected. if this is not the case please create a new issue then let me know more info
